scripts/up.sh

```diff
@@ -1,79 +1,79 @@
 #!/usr/bin/env bash
-#
+
 # Bring up the SDK test environment.
 
 set -e
 
-rootdir=`dirname $0`
-pushd $rootdir/.. > /dev/null
+START=$(date "+%s")
 
 # Defaults
 TYPE_OVERRIDE=""
-DAEMON_FLAG="-d"
-SKIP_BUILD=0
-SHOW_HELP=0
-ENV_FILE=".up-env"
-PARALLEL="--parallel"
-
-show_help() {
-  echo "Manage bringing up the SDK test environment."
-  echo
-  echo "Usage: up.sh [options]"
-  echo
-  echo "Options:"
-  echo "  -f <FILE>  Override the environment file."
-  echo "  -t <TYPE>  Override the installation type specified in the environment file."
-  echo "             Valid types: ['channel', 'type']"
-  echo "  -s         Skip rebuilding the docker image."
-  echo "  -i         Start the docker environment in interactive mode."
-  echo "  -h         Provide this help information."
-  echo "  -p         Disable parallel build mode. Probably only useful when editing the test environment."
-}
-
-# Parse arguments
-while getopts "pf:t:sih" opt; do
-  case "$opt" in
-    f) ENV_FILE=$OPTARG; ;;
-    i) unset DAEMON_FLAG; ;;
-    s) SKIP_BUILD=1; ;;
-    t) TYPE_OVERRIDE=$OPTARG; ;;
-    h) show_help; exit 0 ;;
-    p) unset PARALLEL; ;; 
-  esac
-done
-
-# Verify there are no positional parameters with getopt/getopts
-shift "$((OPTIND-1))"
-if [[ "$1" != "" ]]; then
-  echo "No positional arguments should be provided, found '$@'"
-  echo
-  show_help
-  exit
-fi
+ENV_FILE=".env"
 
-# Load environment.
-source $ENV_FILE
+# Load environment
+echo "up.sh: sourcing environment vars from-->$ENV_FILE"
+source "$ENV_FILE"
 
 # Choose which dockerfile to use.
 TYPE=${TYPE_OVERRIDE:-$TYPE}
 if [[ $TYPE == "channel" ]] || [[ $TYPE == "source" ]]; then
   export TYPE="$TYPE"
 else
-  echo "Unknown environment: $TYPE"
+  echo "up.sh: Unknown environment: $TYPE"
   exit 1
 fi
 
-echo "Bringing up network with '$TYPE' configuration."
+echo "up.sh: Before bootrapping, try cleaning up first..."
 
-# Make sure it isn't running
-./scripts/down.sh
+# Make sure test-sdk sandbox isn't running and clean up any docker detritous
+./scripts/down.sh -f "$ENV_FILE"
+rm -rf "$SANDBOX_DIR"
+echo "up.sh: seconds it took to get to end of $SANDBOX_DIR cleanup: " + $(($(date "+%s") - $START))
 
-# Remove the containers to allow re-running stateful tests.
-docker-compose rm --force
+SANDBOX_CFG="_config.harness"
+cp config.harness "$SANDBOX_CFG"
+if [[ $TYPE == "channel" ]]; then
+  ALGOD_URL=""
+  ALGOD_BRANCH=""
+  ALGOD_SHA=""
+else
+  ALGOD_CHANNEL=""
+fi
 
-# When developing, it's often useful to skip the build phase.
-if [[ $SKIP_BUILD -eq 0 ]]; then
-  docker-compose build --no-cache $PARALLEL
+if [[ $OSTYPE == 'darwin'* ]]; then
+  SED_CMD='sed -i "" '
+else
+  SED_CMD='sed -i '
 fi
 
-docker-compose up $DAEMON_FLAG
+echo "up.sh: On this OSTYPE=$OSTYPE using sed variant $SED_CMD"
+echo "up.sh: apply regex replacements on SANDBOX_CFG=$SANDBOX_CFG"
+$SED_CMD "s|#ALGOD_CHANNEL|$ALGOD_CHANNEL|g" "$SANDBOX_CFG"
```

Comment:
These sed commands are creating a new file called `_config.harness""` on my machine - I don't know enough to figure out why though

Reply:
Oh I guess I might be on `Darwin` - I notice that there are double quotes after `sed -i` depending on the OS

Reply:
yeah... maybe that's how the mac os hack works. I'll be investigating some ideas that Will put up and look into improving the way substitution is working.

Reply:
I've replaced `sed` with `envsubst` so there should be less crud. Please try again.